build.sbt

```diff
@@ -1,92 +1,34 @@
-  .aggregate(
-    `evaluator-server`,
-    `evaluator-shared-jvm`,
-    `evaluator-shared-js`,
-    `evaluator-client-jvm`,
-    `evaluator-client-js`)
-
-lazy val `evaluator-shared` = (crossProject in file("shared"))
-  .enablePlugins(AutomateHeaderPlugin)
-  .settings(name := "evaluator-shared")
-
-lazy val `evaluator-shared-jvm` = `evaluator-shared`.jvm
-lazy val `evaluator-shared-js`  = `evaluator-shared`.js
-
-lazy val `evaluator-client` = (crossProject in file("client"))
-  .dependsOn(`evaluator-shared`)
-  .enablePlugins(AutomateHeaderPlugin)
-  .settings(
-    name := "evaluator-client",
-    libraryDependencies ++= Seq(
-      %%("roshttp"),
-      %%("cats-free"),
-      %%("circe-core"),
-      %%("circe-generic"),
-      %%("circe-parser"),
-      %%("log4s"),
-      %("slf4j-simple"),
-      %%("scalatest") % "test"
-    )
-  )
-  .jsSettings(sharedJsSettings: _*)
-
-lazy val `evaluator-client-jvm` = `evaluator-client`.jvm
-lazy val `evaluator-client-js`  = `evaluator-client`.js
-
 lazy val `evaluator-server` = (project in file("server"))
-  .dependsOn(`evaluator-shared-jvm`)
   .enablePlugins(JavaAppPackaging)
   .enablePlugins(AutomateHeaderPlugin)
   .enablePlugins(sbtdocker.DockerPlugin)
   .enablePlugins(BuildInfoPlugin)
   .settings(noPublishSettings: _*)
   .settings(
     name := "evaluator-server",
-    libraryDependencies ++= Seq(
-      %%("monix"),
-      %%("circe-core"),
-      %%("circe-generic"),
-      %%("circe-parser"),
-      %%("log4s"),
-      %("slf4j-simple"),
-      %%("http4s-dsl", http4sV),
-      %%("http4s-blaze-server", http4sV),
-      %%("http4s-blaze-client", http4sV),
-      %%("http4s-circe", http4sV),
-      %("config"),
-      %%("jwt-core"),
-      "io.get-coursier" %% "coursier" % "1.0.0-M15-3",
-      "io.get-coursier" %% "coursier-cache" % "1.0.0-M15-3",
-      %%("scalatest")   % "test"
-    ),
+    serverHttpDependencies,
     assemblyJarName in assembly := "evaluator-server.jar"
   )
   .settings(dockerSettings: _*)
   .settings(buildInfoSettings: _*)
   .settings(serverScalaMacroDependencies: _*)
 
 lazy val `smoketests` = (project in file("smoketests"))
-  .dependsOn(`evaluator-server` % "compile->compile;test->test")
+  .dependsOn(`evaluator-server`)
   .enablePlugins(BuildInfoPlugin)
   .settings(noPublishSettings: _*)
   .settings(
     name := "evaluator-server-smoke-tests",
-    libraryDependencies ++= Seq(
-      %%("circe-core"),
-      %%("circe-generic"),
-      %%("circe-parser"),
-      %%("http4s-blaze-client", http4sV),
-      %%("http4s-circe", http4sV),
-      %%("jwt-core"),
-      %%("scalatest") % "test"
-    )
+    smoketestDependencies
   )
   .settings(buildInfoSettings: _*)
 
+lazy val root = (project in file("."))
+  .settings(mainClass in Universal := Some("org.scalaexercises.evaluator.EvaluatorServer"))
+  .settings(stage := (stage in Universal in `evaluator-server`).value)
+  .settings(noPublishSettings: _*)
+  .aggregate(`evaluator-server`)
```

Comment:
We should aggregate the smoke tests to also compile from the root module

Reply:
In master the smoketests project wasn't aggregated to root. Maybe can be compiled, but the execution of those tests requires deployment